### PR TITLE
Fold common_py3 back into other files

### PR DESCRIPTION
**Pull request by chadwhitacre**
https://tahoe-lafs.org/trac/tahoe-lafs/ticket/3456

### Todo

- [x] [LoggingServiceParent, ShouldFailMixin](https://github.com/tahoe-lafs/tahoe-lafs/commit/a758f32edfd816ca4d56e797767fdaa4312f5bb1) (common)
- [x] [Marker, FakeCanary](https://github.com/tahoe-lafs/tahoe-lafs/commit/416ab643359c10e80f9ef0b18e8e4d173757bde5)
- [x] [ReallyEqualMixin, skip_if_cannot_represent_filename](https://github.com/tahoe-lafs/tahoe-lafs/commit/e13c0cf4d34dacf00b51c0d788fd859197467ac2)
- [x] [ShouldFailMixin](https://github.com/tahoe-lafs/tahoe-lafs/commit/44143d1b083acf23b93bbfb5eb37a23e9af22934) (common_util, [cf.](https://github.com/tahoe-lafs/tahoe-lafs/commit/d195ae43232b654371d1f3d476fc0b7bde640be9))
- [x] [SignalMixin](https://github.com/tahoe-lafs/tahoe-lafs/commit/bf134019797b2fadc8f92405720b1956e4b2fbc5)
- [x] [TimezoneMixin](https://github.com/tahoe-lafs/tahoe-lafs/commit/51e9c2183cf15268985412481b01112f8b2a5b77)

### CODE REVIEW

File: src/allmydata/test/common_util.py [hunk 1]
```diff
@@ -78,13 +89,114 @@ def flip_one_bit(s, offset=0, size=None):
...
         return d
 
 
+class Marker(object):
+    pass
+
+class FakeCanary(object):
+    """For use in storage tests.
+
+    Can be moved back to test_storage.py once enough Python 3 porting has been
```
Comment (itamarst) [inline src/allmydata/test/common_util.py:133]: Maybe move this to test_storage.py? Unless it's used in multiple storage test modules.

**Review by itamarst — APPROVED** [review]
Maybe do what I said above, otherwise from my perspective this is fine.

File: src/allmydata/test/common_util.py (hunk 1 above)
Comment (chadwhitacre) [reply src/allmydata/test/common_util.py:133]: I am seeing it used in several other test modules:

```
test_storage_web.py
53:from .common_util import FakeCanary
310:        canary = FakeCanary()

web/test_introducer.py
37:    FakeCanary,
197:                FakeCanary(),
208:            FakeCanary(),

common_util.py
130:class FakeCanary(object):

test_storage.py
55:from .common_util import FakeCanary
127:                          FakeCanary())
137:                          FakeCanary())
231:                          FakeCanary())
390:            canary = FakeCanary()
547:        canary = FakeCanary()
580:        canary = FakeCanary(True)
652:        canary = FakeCanary()

test_crawler.py
30:from allmydata.test.common_util import StallMixin, FakeCanary
130:                                              set([0]), 99, FakeCanary())
```

Comment (chadwhitacre) [reply src/allmydata/test/common_util.py:133]: But in that case ... https://github.com/tahoe-lafs/tahoe-lafs/pull/846.

**Pull request merged**